grace/evaluation/process.py

```diff
@@ -28,30 +30,68 @@ def update_graph_with_dummy_predictions(
     Notes
     -----
     - Modifies the graph in place.
-    - Only motifs with the same object identity will be linked.
+    - The 'uncertainty' parameter separates the distributions of predictions
+        for the respective classes, e.g. 0.5 will produce a distribution of
+        attribute values for TN between [0.0, 0.5) & TP between (0.5, 1.0]
+        which are very close, whilst 0.1 will force larger separation between
+        them, i.e. TN between [0.0, 0.1) & TP between (0.9, 1.0].
+    - Only motifs with the same object identity should be linked (TODO)
     """
 
     # Make sure all objects are labelled according to object identity:
     nodes = list(G.nodes.data())
-    obj_identity = ["object_idx" in node for _, node in nodes]
-    assert all(obj_identity)
+    obj_identity = ["object_idx" in n and "label" in n for _, n in nodes]
```

Comment:
Is `object_idx` guaranteed to be in the node?

Reply:
Not necessarily, the code chooses whether the GT labels come from the dummy or real annotations. That's what the assert `or` statement checks - some GT annotation must be present.